### PR TITLE
Reduce hashmap one expansion

### PR DESCRIPTION
**Pull request by xianlaioy**
0.75 is the default load factor,if the maps size equals length need divided by 0.75.

### CODE REVIEW

File: core/src/main/java/io/seata/core/model/BranchStatus.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public int getCode() {
         return code;
     }
 
-    private static final Map<Integer, BranchStatus> MAP = new HashMap<>(values().length);
+    private static final Map<Integer, BranchStatus> MAP = Maps.newHashMapWithExpectedSize(values().length);
```
Comment (l81893521) [inline core/src/main/java/io/seata/core/model/BranchStatus.java:111]: I think that s same and branch status was fixed, it would not let the map resize.

Comment (xianlaioy) [reply core/src/main/java/io/seata/core/model/BranchStatus.java:111]: This way of writing is bound to expand capacity

Comment (l81893521) [reply core/src/main/java/io/seata/core/model/BranchStatus.java:111]: You re right, thanks.

**Comment by xianlaioy** [comment]
trigger a new build

**Comment by l81893521** [comment]
> trigger a new build

check here:
https://travis-ci.com/github/seata/seata/jobs/346914071

**Comment by xianlaioy** [comment]
> > trigger a new build
> 
> check here:
> https://travis-ci.com/github/seata/seata/jobs/346914071

seata-serializer-seata 1.3.0-SNAPSHOT .............. FAILURE

Strangely, I didn't change the code for this module

**Comment by l81893521** [comment]
> > > trigger a new build
> > 
> > 
> > check here:
> > https://travis-ci.com/github/seata/seata/jobs/346914071
> 
> seata-serializer-seata 1.3.0-SNAPSHOT .............. FAILURE
> 
> Strangely, I didn't change the code for this module

maybe you can check the dependency of guava in core/pom.xml

**Comment by xianlaioy** [comment]
> > > > trigger a new build
> > > 
> > > 
> > > check here:
> > > https://travis-ci.com/github/seata/seata/jobs/346914071
> > 
> > 
> > seata-serializer-seata 1.3.0-SNAPSHOT .............. FAILURE
> > Strangely, I didn't change the code for this module
> 
> maybe you can check the dependency of guava in core/pom.xml

tks,

File: core/src/main/java/io/seata/core/model/BranchStatus.java [hunk 2]
```diff
@@ -108,7 +108,7 @@ public int getCode() {
         return code;
     }
 
-    private static final Map<Integer, BranchStatus> MAP = new HashMap<>(values().length);
+    private static final Map<Integer, BranchStatus> MAP = new HashMap<>((int) ((float) values().length / 0.75F + 1.0F));
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/model/BranchStatus.java:111]: The capacity of the map is always 2 ^ n.

Comment (xianlaioy) [reply core/src/main/java/io/seata/core/model/BranchStatus.java:111]: tks，min value is 2^n。So 10 won't expand, if it's 12 it will.

Comment (xianlaioy) [reply core/src/main/java/io/seata/core/model/BranchStatus.java:111]: `(int) ((float) values().length / 0.75F + 1.0F)  `
It should be written in a more correct way

Comment (slievrly) [reply core/src/main/java/io/seata/core/model/BranchStatus.java:111]: I don't recommend that. 0.75 is a hashMap default factor, should not directly write this value here,  if 0.75 becomes a different value in a later JDK version, what shall we do?

Comment (xianlaioy) [reply core/src/main/java/io/seata/core/model/BranchStatus.java:111]: It's supposed to be a constant, in theory it doesn't change。JDK Hashmap Source Code is optimized all the time, but this load factor has not changed

Comment (xianlaioy) [reply core/src/main/java/io/seata/core/model/BranchStatus.java:111]: We encapsulate a method to handle internal logic, and default load factors can also be defined as variables. Forget it. I close it.

**Pull request closed without merging**